### PR TITLE
Community distribution fixes

### PR DESCRIPTION
**Pull request by ramereth**
This applies some fixes which were introduced in [1] that help support community
distributions better. Some other fixes include:

- Ensure we pull in mixlib-install >= 3.11.23 which includes the patches
  introduced in [1]
- Create dist.rb file for renaming chef to something else and also set the SLES
  project versions to other names

[1] https://github.com/chef/mixlib-install/pull/292

Signed-off-by: Lance Albertson <lance@osuosl.org>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by ramereth** [comment]
Actually, I think I may need to push some additional changes so please hold on doing a full review.

File: Gemfile.lock [hunk 1]
```diff
@@ -17,7 +17,7 @@ GEM
     kgio (2.10.0)
     method_source (0.8.2)
     mime-types (1.25.1)
-    mixlib-install (3.11.21)
```
Comment (tas50) [inline Gemfile.lock]: Let's make sure to get a pin the gemfile to >= the necessary version so we don't have some weird depsolving mess that reverts this later on. It's happened more than once in various projects.

Comment (ramereth) [reply Gemfile.lock]: Done!

**Comment by tas50** [comment]
@ramereth looks like you're missing the actual dist file and any usage.

**Comment by ramereth** [comment]
@tas50 how does it look now?

File: Gemfile [hunk 2]
```diff
@@ -11,7 +11,7 @@ gem "rest-client"
 gem 'rake'
 gem 'redis'
 gem 'mixlib-versioning', '~> 1.1.0'
-gem 'mixlib-install'
+gem 'mixlib-install', '>= 3.11.23'
```
Comment (jaymalasinha) [inline Gemfile]: Why do we need to pin mixlib-install here ?  Latest version of mixlib-install is "3.12.1"  which is already pulled in last update

Comment (ramereth) [reply Gemfile]: See: https://github.com/chef/omnitruck/pull/420#discussion_r357954759

**Comment by tas50** [comment]
Thanks for putting this together @ramereth

**Pull request merged**